cpp/dynamic_community_detection_module/algorithm/dynamic_community_detection.cpp

```diff
@@ -0,0 +1,370 @@
+
+  auto neighbor_set = this->is_directed
+                          ? graph->InNeighbours(graph->GetInnerNodeId(node_id))
+                          : graph->Neighbours(graph->GetInnerNodeId(node_id));
+  for (const auto node_i : neighbor_set) {
+    neighbors.insert(graph->GetMemgraphNodeId(node_i.node_id));
+  }
+
+  return neighbors;
+}
+
+std::vector<std::uint64_t> LabelRankT::NodesMemgraphIDs() {
+  std::vector<std::uint64_t> nodes;
+
+  for (const auto node : graph->Nodes()) {
+    nodes.push_back(graph->GetMemgraphNodeId(node.id));
+  }
+
+  return nodes;
+}
+
+void LabelRankT::SetStructures(std::uint64_t node_id) {
+  times_updated[node_id] = 0;
+  std::unordered_map<std::uint64_t, double> node_label_Ps;
+
+  double sum_w_i = w_selfloop;
+  for (const auto node_j_id : InNeighborsMemgraphIDs(node_id)) {
+    sum_w_i += GetTotalWeightBetween(node_j_id, node_id);
+  }
+
+  // add self-loop
+  node_label_Ps[node_id] = w_selfloop / sum_w_i;
+
+  // add other edges
+  for (const auto node_j_id : InNeighborsMemgraphIDs(node_id)) {
+    node_label_Ps[node_j_id] +=
+        GetTotalWeightBetween(node_j_id, node_id) / sum_w_i;
+  }
+
+  label_Ps[node_id] = node_label_Ps;
+  sum_w[node_id] = sum_w_i;
+}
+
+void LabelRankT::RemoveDeletedNodes(
+    std::unordered_set<std::uint64_t>& deleted_nodes_ids) {
+  for (const auto node_id : deleted_nodes_ids) {
+    label_Ps.erase(node_id);
+    sum_w.erase(node_id);
+    times_updated.erase(node_id);
+  }
+}
+
+double LabelRankT::GetTotalWeightBetween(std::uint64_t from_node_id,
+                                         std::uint64_t to_node_id) {
+  double total_weight = 0;
+  for (auto edge_id :
+       graph->GetEdgesBetweenNodes(graph->GetInnerNodeId(from_node_id),
+                                   graph->GetInnerNodeId(to_node_id))) {
+    auto edge = graph->GetEdge(edge_id);
+
+    // edge direction check
+    if (this->is_directed &&
+        (graph->GetMemgraphNodeId(edge.from) != from_node_id ||
+         graph->GetMemgraphNodeId(edge.to) != to_node_id))
+      continue;
+
+    total_weight += GetWeight(edge_id);
+  }
+
+  return total_weight;
+}
+
+double LabelRankT::GetWeight(std::uint64_t edge_id) {
+  if (!this->is_weighted) return this->DEFAULT_WEIGHT;
+
+  return graph->GetWeight(edge_id);
+}
+
+void LabelRankT::ResetTimesUpdated() {
+  for (auto [node_id, _] : times_updated) {
+    times_updated[node_id] = 0;
+  }
+}
+
+std::int64_t LabelRankT::NodeLabel(std::uint64_t node_id) {
+  std::int64_t node_label = -1;
+  double max_P = 0;
+
+  for (const auto [label, P] : label_Ps[node_id]) {
+    if (P > max_P || (P == max_P && (std::int64_t)label < node_label)) {
+      max_P = P;
+      node_label = label;
+    }
+  }
+
+  return node_label;
+}
+
+std::unordered_map<std::uint64_t, std::int64_t> LabelRankT::AllLabels() {
+  std::unordered_map<std::uint64_t, std::int64_t> labels;
+
+  std::set<std::int64_t> labels_ordered;
+  std::map<std::int64_t, std::int64_t> lookup;
```

Comment:
```suggestion
  std::unordered_map<std::int64_t, std::int64_t> lookup;
```
Why not `unordered_map`?

Reply:
Fair enough!